src/server/pps/server/pipeline_controller.go

```diff
@@ -188,48 +190,34 @@ func (pc *pipelineController) step(timestamp time.Time) (retErr error) {
 		tracing.TagAnySpan(span, "pollpipelines-event", "true")
 	}
 	defer func() {
-		tracing.FinishAnySpan(span, "err", retErr)
+		tracing.FinishAnySpan(span, "err", finalErr)
 	}()
 
-	// query pipelineInfo
-	var pi *pps.PipelineInfo
-	var err error
-	if pi, err = pc.tryLoadLatestPipelineInfo(); err != nil && collection.IsErrNotFound(err) {
-		// if the pipeline info is not found, interpret the operation as a delete
+	// if we fail to create a new step, there was an error querying the pipeline info, and there's nothing we can do
+	step, err := pc.newStep()
+	if err != nil {
+		finalErr = err
+		log.Errorf("PPS master: failed to set up step data to handle event for pipeline '%s': %v", pc.pipeline,
+			errors.Wrapf(err, "failing pipeline %q", pc.pipeline))
```

Comment:
I may be misreading this, but I think there may be a bug here, where the log line says that it's failing the pipeline, but the pipeline doesn't get failed. If it makes sense to move this into the retry loop with `getRC`, that could be one fix, but we could also call `setPipelineFailure` in two places.

Reply:
You're right. I updated the message. There's actually nothing we can do in this case, since we weren't able to retrieve the pipelineInfo from the DB. We can only log the error